inspire/modules/deposit/static/js/deposit/literature_submission_form.js

```diff
@@ -796,6 +804,48 @@ define(function(require, exports, module) {
       });
     },
 
+    /**
+     * Enable or disable fields
+     */
+    toggleDisableFields: function toggleDisableFields(fields, toggle) {
+      $.each(fields, function(index, field) {
+        $(field).attr('disabled', toggle);
+      });
+    },
+
+    /**
+     * Adds or removes temporary message on top of parent box
+     */
+    toggleTemporaryMessage: function toggleTemporaryMessage(parentBox, temp_message) {
+      if (temp_message === undefined) {
+        $(parentBox).parents('.panel-body').children('.alert').remove();
+      } else {
+        var tpl_loading_message = require('hgn!js/deposit/templates/loading_message');
+        $(parentBox).parents('.form-group').before(tpl_loading_message({message: temp_message}));
+      }
+    },
+
+    /**
+     * Imports fields from given API and fills the deposit form using fillForm()
+     *
+     * @param options {} dictionary with 'url' for ajax call and its 'params'
+     */
+    fillFormExtra: function fillFormExtra(options) {
+      this.toggleTemporaryMessage(options.blockFields[0], options.message);
+      this.toggleDisableFields(options.blockFields, true);
+
+      var that = this;
+
+      $.ajax({
+        url: options.url,
+        data: options.params
+      }).done(function(result) {
+        that.fillForm(result);
```

Comment:
If the ``result`` here is just the journal information fields, will calling ``fillForm`` not have undesired side effects?

That function does [many things](https://github.com/inspirehep/inspire-next/blob/master/inspire/modules/deposit/static/js/deposit/literature_submission_form.js#L723) including recreating the author fields (removing them first).

So I guess this second call to fillForm will delete the authors?